### PR TITLE
Search field is now always aligned left and cross aligned right.

### PR DESCRIPTION
**Pull request by blattmann**
Must fix: #707, #708, #709, #710

Search field and cross adjusted. Only 2 modes available now:
* Mobile view < 768px
* Tablet view > 768px

Example:
https://blattmann.github.io/ngeo/707_search_field_alignments/examples/contribs/gmf/apps/mobile/

### CODE REVIEW

File: contribs/gmf/less/mobile.less [hunk 1]
```diff
@@ -118,7 +118,7 @@ span.twitter-typeahead {
     position: absolute;
     top: 0;
     left: 50%;
-    margin-left: ~"calc("@search-input-size / 2 ~" - " @map-tools-size~")";;
+    margin-left: ~"calc("@search-input-size-clear / 2 ~" - " @map-tools-size~")";
```
Comment (ger-benjamin) [inline contribs/gmf/less/mobile.less]: The `~"..." ` is used to say to "less" to not interpret this content. That means that the browser will do the calculation. When we can, we prefer that less do the calculation, so our application is a little bit faster.

In this case, all units are in px (the "2" can be in pixel or other thing, that doesn't change the result) so less can do the complete job and I pretty sur that we can write this line like this:
`margin-left: calc(@search-input-size-clear / 2 - @map-tools-size);`

That's easier to read and faster for the browser.

**Comment by blattmann** [comment]
@ger-benjamin :
Ok, if we do not need to calculate this value in the browser, we do not need to escape this line.
Regarding the performance we should use then:

`margin-left: ceil(@search-input-size / 2 - @map-tools-size);`

This produces a calculated value. E.g. `margin-left: 4em;` instead of `margin-left: calc(4em);`

What do you think?

**Comment by ger-benjamin** [comment]
There is a lot of "min-device-width: @screen-sm-min" or max-device-width: @screen-sm-max rules.
That's hard to understand. 

Can you just move and regroups your code to have:
 - A default part first (for mobile but it's the default, we don't need @"screen-sm-..." rules)
- A part only for the tablette, starting with "@media (min-width: @screen-sm-min) {" and that overrides css rules that must be overrided (no needs of a rule for landscape and a rule for portrait, that's the same behaviors, so only one rule for tablette, at the end of this file).

(quick example, not working for anything)

```
input {
  width: @search-input-size;
  padding: 0 5px;
  border-width: 0;
  border-radius: 0;
  height: @map-tools-size - .2em;
}
/** ... **/

// For tablet only
@media (min-width: @screen-sm-min) {
 input {
    width: 100%
  }
}
/** ... **/
```

**Comment by ger-benjamin** [comment]
> This produces a calculated value. E.g. margin-left: 4em; instead of margin-left: calc(4em);

That's a good idea !

**Comment by ger-benjamin** [comment]
Looks good on the apps/mobile but please, take the time to understand my two comments and address them.

**Comment by eleu** [comment]
note: addresses #707

**Comment by ger-benjamin** [comment]
Clean on #743

**Pull request closed without merging**